### PR TITLE
Fix #1882: Added test for RecentIssues Component

### LINKED ISSUE #1882: Add tests for <RecentIssues> component
Write unit tests for the `<RecentIssues>` React component to ensure expected behavior, edge case handling, and rendering logic.

## Essential Test Coverage Checklist

- [ ] **Renders successfully with minimal required props**  
- [ ] **Conditional rendering logic**  
- [ ] **Prop-based behavior** – different props affect output  
- [ ] **Event handling** – simulate user actions and verify callbacks  
- [ ] **State changes / internal logic**  
- [ ] **Default values and fallbacks**  
- [ ] **Text and content rendering**  
- [ ] **Handles edge cases and invalid inputs**  
- [ ] **Accessibility roles and labels**  
- [ ] **DOM structure / classNames / styles**

## Test Reference  
You can refer to the `AutoScrollToTop.test.tsx` file for an example of structure and best practices.  
To explore more examples, see the full component tests folder.

### PR DESCRIPTION
**Pull request by anurag2787**
Resolves #1882 

# Description
Added Unit Tests for RecentIssues Component
Added comprehensive test suite with 18 test cases covering:

- Renders successfully with minimal required props
- Conditional rendering logic
- Prop-based behavior – different props affect output
- Event handling – simulate user actions and verify callbacks
- State changes / internal logic
- Default values and fallbacks
- Text and content rendering
- Handles edge cases and invalid inputs
- Accessibility roles and labels
- DOM structure / classNames / styles

## Checklist

- [x] I've read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md).
- [x] I've run `make check-test` locally; all checks and tests passed.

### CODE REVIEW

**Comment by anurag2787** [comment]
Hy @kasya I ran make check-test on my side and the failing test is in __tests__/unit/pages/Home.test.tsx. It doesn’t appear to be related to my changes in RecentIssues.test.tsx.

Screenshot from 2025-08-19 10-17-19

File: frontend/__tests__/unit/components/RecentIssues.test.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    const issue: Issue = { ...baseIssue, author: undefined }
+    render(<RecentIssues data={[issue]} />)
+    expect(screen.getByText('Issue Title')).toBeInTheDocument()
+  })
+
+  it('handles edge case: missing title', () => {
+    const issue: Issue = { ...baseIssue, title: undefined }
+    render(<RecentIssues data={[issue]} />)
+    expect(screen.getByText('Recent Issues')).toBeInTheDocument()
+  })
```
[the remark this answers is not in the document]
Comment (anurag2787) [reply frontend/__tests__/unit/components/RecentIssues.test.tsx]: as if we want the test to expect a fallback like untitled then we also need to update the ItemCardList component which i don't think required here
